README.md

```diff
@@ -288,6 +288,37 @@ public class FailoverSampleApp2 {
   }
 }
 ```
+
+### Plugin Manager
+
+<div style="text-align:center"><img src="./docs/files/images/plugin_manager_diagram.png" /></div>
+
+Figure shows a simplified workflow of the plugin manager. Starting at the top, when a new query is executed by the application, it is passed into the plugin manager. From the  plugin manager, the query is passed in order of plugins loaded, in this figure, `Custom Plugin A` to `Custom Plugin B` and finally to `Default Plugin` which executes the query and returns the result sets back up the chain.
+Custom plugins can be used to interact with queries, such as adding a feature to monitor a connection's status in the case of `NodeMonitoringConnectionPlugin`. 
+
+By default, `NodeMonitoringConnectionPlugin` is loaded for [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc#aws-iam-database-authentication). Additional custom plugins can be included and chained together.
+
+To learn how to write custom plugins, refer to examples located inside [Custom Plugins Demo](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/demo/customplugins).
+
+**Note**: The order of which plugins load in matters.
+
+#### Plugin Manager Parameters
+| Parameter       | Value           | Required      | Description  |
+| ------------- |:-------------:|:-------------:| ----- |
+|`connectionPluginFactories` | String | No | String of custom plugins to use monitoring. If left unset, `NodeMonitoringConnectionPluginFactory` will be loaded. <br/><br/>Each factory in the string should be separated by `,`<br/><br/>NOTE: The order of factories declared matters.  <br/><br/>Example `MyPluginA`, `MyPluginA,MyPluginB,NodeMonitoringConnectionPluginFactory` <br/><br/>**Default value:** `NodeMonitoringConnectionPluginFactory` |
+
+### Enhanced Failure Monitoring
```

Comment:
I thought the feature is called `Enhanced Instance Monitoring`. Can we confirm now so we stay consistent? @sergiyv-bitquill

Reply:
Agree, we need a consistent name

Reply:
I think Enhanced Failure Monitoring is more obvious, `instance` might not be immediately clear. Should we stick to `Enhanced Failure Monitoring`?

Reply:
Can we add an additional column for default values instead of putting them in description?

Reply:
While I understand that we want to keep our examples short and concise, we should aim to provide working examples as much as we can because it would be a better user experience. As in if we tell users they need to provide the `String of fully-qualified class name of plugin factories. `, our example should use fully-qualified class names.

`customplugins.MethodCountConnectionPluginFactory,customplugins.ExecutionTimeConnectionPluginFactory,com.mysql.cj.jdbc.ha.ca.plugins.NodeMonitoringConnectionPluginFactory`

Reply:
I would just put `com.mysql.cj.jdbc.ha.ca.plugins.NodeMonitoringConnectionPluginFactory` for the default value.